init.lua

```diff
@@ -107,14 +107,15 @@ function irc:connect()
 			port = self.config.port,
 			password = self.config.password,
 			timeout = self.config.timeout,
+			reconnect = self.config.reconnect,
 			secure = self.config.secure
 		})
 	end)
 
 	if not good then
 		minetest.log("error", ("IRC: Connection error: %s: %s -- Reconnecting in ten minutes...")
```

Comment:
`s/reconnect/self.config.reconnect/` (or, better, cache it in the function).  But otherwise, yes.